src/components/rating/rating.tsx

```diff
@@ -3,27 +3,38 @@ import {
   Element,
   Event,
   EventEmitter,
-  Fragment,
+  FunctionalComponent,
   h,
-  Listen,
+  Host,
   Method,
   Prop,
   State,
-  VNode
+  Watch
 } from "@stencil/core";
 import { guid } from "../../utils/guid";
 import { Scale } from "../interfaces";
 import { LabelableComponent, connectLabel, disconnectLabel } from "../../utils/label";
 import { connectForm, disconnectForm, FormComponent, HiddenFormInputSlot } from "../../utils/form";
 import { TEXT } from "./resources";
 import { InteractiveComponent, updateHostInteraction } from "../../utils/interactive";
-import { isActivationKey } from "../../utils/key";
 import {
   setUpLoadableComponent,
   setComponentLoaded,
   LoadableComponent,
   componentLoaded
 } from "../../utils/loadable";
+import { Star, StarIconProps } from "./interfaces";
+
+const StarIcon: FunctionalComponent<StarIconProps> = ({ icon, scale, partial }) => (
```

Comment:
Actually, a functional component isn't really necessary here as the only thing that is really changing is the class name. The class name could just be applied on the `calcite-icon` component. I think the functional component would be more useful if it was setting the star icon based on a `full: boolean` variable?